### PR TITLE
Clearable typeahead + npm updates + remove enzyme

### PR DESCRIPTION
**Pull request by iaincrawf**
- upgrade npm packages, including react to 17 
- remove enzyme completely
- rewrite tests without enzyme
- offer built in clear button with typeahead, as used on impbooks and parcel ut
image

### CODE REVIEW

**Comment by iaincrawf** [comment]
If anyone approves this can you also remind me how we release the components lib? I'm sure it's "npm something" but can't remember what

File: src/components/__tests__/SnackbarMessage.test.js [hunk 1]
```diff
@@ -1,27 +1,20 @@
...
-            onClose: jest.fn()
-        };
-    });
-
-    it('should render snackbar', () => {
-        wrapper = mount(
+        render(
             <MuiThemeProvider theme={createTheme()}>
-                <SnackbarMessage {...props} />{' '}
+                <SnackbarMessage  onClose={jest.fn()} message={'Snackbar message'} visible={true} />{' '}
```
Comment (lewisrenfrew) [inline src/components/__tests__/SnackbarMessage.test.js:12]: can probs remove {' '}

File: src/setupTests.js [hunk 2]
```diff
@@ -1,6 +0,0 @@
-const Enzyme = require('enzyme');
```
Comment (lewisrenfrew) [inline src/setupTests.js:1]: is this file just empty now? Can delete if so

**Review by lewisrenfrew — APPROVED** [review]
the CI will publish the package automatically if all the tests pass when you merge to master

File: src/setupTests.js (hunk 2 above)
Comment (iaincrawf) [reply src/setupTests.js:1]: I think I have, it says deleted on the git diff for me! Also didn't realise we had CI on this now, that's handy 😃

**Pull request merged**